firestore/testapp/Assets/Firebase/Sample/Firestore/UIHandler.cs

```diff
@@ -199,12 +199,61 @@ class WaitForTaskCompletion : CustomYieldInstruction {
       }
     }
 
-    protected internal FirebaseFirestore db {
+    protected internal FirebaseFirestore TestFirestore {
       get {
-        return FirebaseFirestore.DefaultInstance;
+        FirebaseFirestore firestore = FirebaseFirestore.DefaultInstance;
+        SetTargetBackend(firestore);
+        return firestore;
       }
     }
 
+    protected internal FirebaseFirestore TestFirestoreWithCustomApp(FirebaseApp app) {
```

Comment:
This method could just be named `TestFirestore`, similar to how API is just `GetInstance`.